project-base/kubernetes/deployments/rabbitmq.yml

```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+    name: rabbitmq
+    labels:
+        app: rabbitmq
+spec:
+    replicas: 1
+    template:
+        metadata:
+            name: rabbitmq
+            labels:
+                app: rabbitmq
+        spec:
+            containers:
+                -   name: rabbitmq
+                    image: rabbitmq:3.7-management-alpine
+                    imagePullPolicy: IfNotPresent
+            restartPolicy: Always
+    selector:
```

Comment:
We tend to keep the `selector` between `replicas` and `template` - please, try to keep things consistent.

Also, I've noticed that some of our manifests (`adminer`, `redis`, `redis-admin`, `selenium-server` and `smtp-server`) don't specify `spec.selector`, which [shouldn't be possible according to the docs](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector). Do you have any idea how Kubernetes behave when no selector is provided?

Reply:
I moved selector between `replicas` and `template` in bffa63ae9a2d508978602f29638afdc8be542656

When I show deployment for redis with `kubectl describe deployments redis`, I see 
```
Name:                   redis
Labels:                 app=redis
Annotations:            deployment.kubernetes.io/revision: 1
...
Selector:               app=redis
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
...
Pod Template:
  Labels:  app=redis
...
```

So it's seems like it's somehow works even without selectors for the other pods.

Reply:
I was going to ask you to create an issue/PR to make it consistent, but it would require writing upgrade notes... Not sure whether it's worth the effort.

I'll let you decide :slightly_smiling_face:

Reply:
It's actually part of a bigger problem. Manifests should be updated to the latest versions and reviewed. I created the issue #1277 to address this.

Reply:
Thanks!